File: Java Advanced/3. Sets and Maps/Sets and Maps/src/LegendaryFarming.java
import java.util.*;
import java.util.stream.Collectors;

public class LegendaryFarming {
    public static void main(String[] args) {
        Scanner s = new Scanner(System.in);
        Map<String, Integer> resources = new TreeMap<>();
        resources.put("shards", 0);
        resources.put("fragments", 0);
        resources.put("motes", 0);
        Map<String, Integer> junk = new TreeMap<>();
        boolean isShadowmourne = false;
        boolean isValanyr = false;
        boolean isDragonwrath = false;
        String shards = "shards";
        String fragments = "fragments";
        String motes = "motes";

        System.out.println();
        while (isDragonwrath == false && isShadowmourne == false && isValanyr == false) {
            String[] line = s.nextLine().toLowerCase().split("\\s+");
            List<String> tokens = new ArrayList<>();
            Collections.addAll(tokens, line);

            for (int i = 0; i < tokens.size(); i++) {
                int qunatity = Integer.parseInt(tokens.get(i));
                tokens.remove(i);
                String material = tokens.get(i);
                tokens.remove(i);
                i--;

                if (resources.containsKey(material)) {
                    int oldQuantity = resources.get(material);
                    resources.put(material, qunatity + oldQuantity);
                } else {
                    if (!junk.containsKey(material)) {
                        junk.put(material, qunatity);
                    } else {
                        int oldQunatity = junk.get(material);
                        junk.put(material, qunatity + oldQunatity);
                    }
                }
                int tempQuantity = 0;
                if (resources.get(shards) >= 250) {
                    isShadowmourne = true;
                    System.out.println("Shadowmourne obtained!");
                    tempQuantity = resources.get(shards);
                    resources.put(shards, tempQuantity - 250);
                    break;
                } else if (resources.get(fragments) >= 250) {
                    isValanyr = true;
                    System.out.println("Valanyr obtained!");
                    tempQuantity = resources.get(fragments);
                    resources.put(fragments, tempQuantity - 250);
                    break;
                } else if (resources.get(motes) >= 250) {
                    isDragonwrath = true;
                    System.out.println("Dragonwrath obtained!");
                    tempQuantity = resources.get(motes);
                    resources.put(motes, tempQuantity - 250);
                    break;
                }

            }
        }

        Map<String, Integer> sortedResources = resources.entrySet().stream()
                .sorted((a, b) -> Integer.compare(b.getValue(), a.getValue()))
                .collect(Collectors.toMap(Map.Entry::getKey, Map.Entry::getValue, (a, b) -> a, LinkedHashMap::new));
        for (Map.Entry<String, Integer> entry : sortedResources.entrySet()) {
            System.out.printf("%s: %d%n", entry.getKey(), entry.getValue());
        }
        for (Map.Entry<String, Integer> entry : junk.entrySet()) {
            System.out.printf("%s: %d%n", entry.getKey(), entry.getValue());
        }


    }
}

File: soft uni prgramming fundamentals/7. Dictionaries, Lambda Expressions and LINQ/Dictionaries/LegendaryFarming/LegendaryFarming.cs
using System;
using System.Collections.Generic;
using System.Linq;

namespace LegendaryFarming
{
    class LegendaryFarming
    {
        static void Main(string[] args)
        {
           
            Dictionary<string, int> resources = new Dictionary<string, int>();
            resources.Add("shards", 0);
            resources.Add("fragments", 0);
            resources.Add("motes", 0);
            Dictionary<string, int> junk = new Dictionary<string, int>();
            bool isShadowmourne = false;
            bool isValanyr = false;
            bool isDragonwrath = false;
            while(isShadowmourne == false && isValanyr == false && isDragonwrath == false)
            {
                List <string>input = Console.ReadLine().ToLower().Split().ToList();
                for (int i = 0; i < input.Count; i++)
                {
                    int quantity = int.Parse(input[i]);
                    input.RemoveAt(i);
                    string material = input[i];
                    input.RemoveAt(i);
                    i--;
                    if (resources.ContainsKey(material)){
                        resources[material] += quantity;
                    }
                    else
                    {
                        if (junk.ContainsKey(material) == false)
                        {
                            junk.Add(material, quantity);
                        }
                        else
                        {
                            junk[material] += quantity;
                        }
                    }
                    if (resources["shards"] >= 250)
                    {
                        isShadowmourne = true;
                        resources["shards"] -= 250;
                        break;
                    }
                    else if (resources["fragments"] >= 250)
                    {
                        isValanyr = true;
                        resources["fragments"] -= 250;
                        break;
                    }
                    else if(resources["motes"] >= 250)
                    {
                        isDragonwrath = true;
                        resources["motes"] -= 250;
                        break;
                    }
                }
            }
            if (isShadowmourne)
            {
                Console.WriteLine("Shadowmourne obtained!");
            }else if (isValanyr)
            {
                Console.WriteLine("Valanyr obtained!");
            }
            else
            {
                Console.WriteLine("Dragonwrath obtained!");
            }
            foreach (var item in resources.OrderBy(z => z.Key).OrderByDescending(x=>x.Value))
            {
                Console.WriteLine($"{item.Key}: {item.Value}");
            }
            foreach (var item in junk.OrderBy(x=>x.Key))
            {
                Console.WriteLine($"{item.Key}: {item.Value}");
            }

        }
    }
}
